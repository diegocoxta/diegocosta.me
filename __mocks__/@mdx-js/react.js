module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
