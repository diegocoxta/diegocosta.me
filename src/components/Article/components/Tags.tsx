import React from 'react';
import styled from 'styled-components';
import kebabCase from 'lodash.kebabcase';

import { Link } from '~/utils/i18n';

const TagList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
`;

const TagItem = styled.li`
  list-style: none;
  padding: 10px 10px 0 0;
  margin-bottom: 5px;
`;

const TagLink = styled(Link)`
  text-decoration: none;
  box-shadow: none;
  font-size: 14px;
  color: ${({ theme }) => theme.accentColor};
  font-weight: 700;
  text-transform: lowercase;

  :hover,
  :focus {
    border-bottom: 1px solid ${({ theme }) => theme.accentColor};
    outline: none;
  }
`;

export interface TagsProps {
  tags?: string[] | null;
}

export default function Tags(props: TagsProps) {
  return (
    <TagList data-testid="article-header-tags">
      {props.tags?.map((tag: string, index: number) => (
        <TagItem key={`${index}-${tag}`} data-testid="article-header-tag">
          <TagLink to={`/tags/${kebabCase(tag ?? '')}`}>#{kebabCase(tag ?? '')}</TagLink>
        </TagItem>
      ))}
    </TagList>
  );
}
