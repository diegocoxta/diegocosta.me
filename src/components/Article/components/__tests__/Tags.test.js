import React from 'react';
import { render } from '@testing-library/react';

import Tags from '../Tags';

describe('<Tags />', () => {
  it('renders properly', () => {
    const { baseElement, getByTestId, getAllByTestId, getByText } = render(<Tags tags={['jest', 'testing library']} />);
    expect(getByTestId('article-header-tags')).toBeTruthy();
    expect(getAllByTestId('article-header-tag').length).toEqual(2);
    expect(getByText('#testing-library')).toBeTruthy();
    expect(getByText('#jest')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders properly without tags', () => {
    const { baseElement, getByTestId, queryAllByTestId } = render(<Tags />);
    expect(getByTestId('article-header-tags')).toBeTruthy();
    expect(queryAllByTestId('article-header-tag').length).toEqual(0);
    expect(baseElement).toMatchSnapshot();
  });
});
