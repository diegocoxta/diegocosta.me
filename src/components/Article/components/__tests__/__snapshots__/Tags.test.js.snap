// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
}

.c1 {
  list-style: none;
  padding: 0 10px 0 0;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-shadow: none;
  font-size: 16px;
  font-weight: 700;
  text-transform: lowercase;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid;
  outline: none;
}

<body>
  <div>
    <ul
      class="c0"
      data-testid="article-header-tags"
    >
      <li
        class="c1"
        data-testid="article-header-tag"
      >
        <a
          class="c2"
          href="/tags/jest"
        >
          #
          jest
        </a>
      </li>
      <li
        class="c1"
        data-testid="article-header-tag"
      >
        <a
          class="c2"
          href="/tags/testing-library"
        >
          #
          testing-library
        </a>
      </li>
    </ul>
  </div>
</body>
`;

exports[`<Tags /> renders properly without tags 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
}

<body>
  <div>
    <ul
      class="c0"
      data-testid="article-header-tags"
    />
  </div>
</body>
`;
